pkgs/build-support/rust/build-rust-crate.nix

```diff
@@ -297,9 +297,15 @@ in
 crate_: lib.makeOverridable ({ rust, release, verbose, features, buildInputs, crateOverrides }:
 
 let crate = crate_ // (lib.attrByPath [ crate_.crateName ] (attr: {}) crateOverrides crate_);
+    processedAttrs = [
+      "src" "buildInputs" "crateBin" "crateLib" "libName" "libPath"
+      "buildDependencies" "dependencies" "features"
+      "crateName" "version" "build" "authors" "colors"
+    ];
+    extraDerivationAttrs = lib.filterAttrs (n: v: ! lib.elem n processedAttrs) crate;
     buildInputs_ = buildInputs;
 in
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (rec {
```

Comment:
the following can be also simplified then:

```diff
- buildInputs = [ rust ncurses ] ++ (crate.buildInputs or []) ++ buildInputs_;
+ buildInputs = [ rust ncurses ] ++ buildInputs_;
```

Reply:
As is, I don't think this is possible. If the `crateOverrides` includes `buildInputs`, that will merged on top of the attribute passed set to `mkDerivation` then the rust and ncurses inputs will be lost. We could rename the attribute in crate overrides to be `extraBuildInputs` to be less surprising, or use a different composition that lets `crateOverrides` read more of the context it's updating, so it can explicitly append or replace the value as appropriate.

Reply:
Forgot about that. You are right.